modules/diskq/qdisk.c

```diff
@@ -182,6 +187,55 @@ _create_file(const gchar *filename, gint *fd)
   return TRUE;
 }
 
+static gboolean
+_grab_filename_locks(const gchar *dir, gint *fd)
+{
+  gchar *flock_file_path = g_strdup_printf("%s/%s", dir, FLOCK_FILENAME);
+
+  if (!_create_path(flock_file_path))
+    {
+      msg_error("Error creating dir for disk-buffer flock file",
+                evt_tag_str("filename", flock_file_path),
+                evt_tag_error("error"));
+      g_free(flock_file_path);
+      return FALSE;
+    }
+
+  g_mutex_lock(&filename_lock);
+
+  *fd = open(flock_file_path, O_RDONLY | O_CREAT, 0600);
+  if (*fd < 0)
+    {
+      msg_error("Failed to open disk-buffer flock file",
+                evt_tag_str("filename", flock_file_path),
+                evt_tag_error("error"));
+      g_mutex_unlock(&filename_lock);
+      g_free(flock_file_path);
+      return FALSE;
+    }
+
+  if (flock(*fd, LOCK_EX) < 0)
```

Comment:
`flock` is not completely portable, but it seems to be working on our supported platforms.

Reply:
Thanks for checking!